src/api/c/internal.cpp

```diff
@@ -168,3 +168,32 @@ af_err af_is_owner(bool *result, const af_array arr)
     CATCHALL;
     return AF_SUCCESS;
 }
+
+af_err af_get_allocated_bytes(size_t *bytes, const af_array arr)
+{
+    try {
+        af_dtype ty = getInfo(arr).getType();
+
+        size_t res = 0;
+
+        switch (ty) {
+        case f32: res = getArray<float  >(arr).getDataDims().elements() * sizeof(float);            break;
```

Comment:
Can you move this into the backend? This type of functionality should not be exposed in the API layer.